### PR TITLE
Clean up and use framework

### PR DESCRIPTION
**Pull request by GeneTinderholm**

### CODE REVIEW

File: src/glue_biscuit/schema_utils.py [hunk 1]
```diff
@@ -182,20 +180,27 @@ def wrapper_func(*args, **kwargs):
...
-    spark = SparkSession.builder.getOrCreate()
-    glue_context = GlueContext(spark.sparkContext)
+def __find_glue_context(*args, **kwargs) -> GlueContext:
+    for arg in args:
+        if isinstance(arg, GlueContext):
+            return arg
+    for k, arg in kwargs:
+        if isinstance(arg, GlueContext):
+            return arg
+
```
Comment (GeneTinderholm) [inline src/glue_biscuit/schema_utils.py:197]: I didn't want to force anyone to use anything in any particular position, but if this is too much magic, let me know.

File: src/simple_job/test_load_books.py [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
 from simple_job.load_books import main, load_books, save_books
 
 
+@pytest.fixture(autouse=True)
+def mock_schema_from_glue():
+    with patch("simple_job.load_books.schema_from_glue") as mock_sink_func:
+        mock_sink_func.side_effect = lambda *args: schema_from_cloudformation('./template.yml', args[1])
+        yield mock_sink_func
+
+
```
Comment (GeneTinderholm) [inline src/simple_job/test_load_books.py:22]: This worked out rather nicely after adding a requirement for the glue context to be passed in. The end user only had to end up mocking their own schema func.

Comment (GeneTinderholm) [reply src/simple_job/test_load_books.py:22]: (or, rather, the things they used inside it)

File: src/glue_biscuit/schema_utils.py (hunk 1 above)
Comment (prowe) [reply src/glue_biscuit/schema_utils.py:197]: Seems fine.. Maybe should add a `raise` at the end explaining the need for a `GlueContext`

**Review by prowe — APPROVED** [review]

**Pull request merged**